datastore/gcloud/aio/datastore/datastore.py

```diff
@@ -37,7 +39,36 @@
 log = logging.getLogger(__name__)
 
 
-class Datastore:
+class DatastoreMeta(type):
+    _entity_result_kind = None
+    _key_kind = None
+
+    @property
+    def entity_result_kind(cls):
+        return cls._entity_result_kind
+
+    @entity_result_kind.setter
+    def entity_result_kind(cls, value):
```

Comment:
Functions added should have type hints so `mypy` can validate in CI.

Reply:
Yes. This one, in particular, is difficult because `value` is a type that contains an attribute named `entity_kind` that is also a type.